### PR TITLE
Parse output from "unparseable error" from black

### LINKED ISSUE #399: Output from 'nbqa black' has line numbers referring to Python script, not Jupyter notebook
I was testing the command

```Python
nbqa black notebook.ipynb --nbqa-mutate
```

from [here](https://github.com/nbQA-dev/nbQA/issues/289#issuecomment-712199396) and while figuring out dependencies (black was missing) ended up having that command in a code cell in the active notebook I wanted to test the mutation on, with the path to the notebook (stated as ```correctpath``` below) originally mis-specified (stated as ```wrongpath``` below). 

When calling the command (from a terminal) on that notebook, with the path specified correctly, I got the error message

```Python
error: cannot format correctpath/notebook.ipynb: Cannot parse: 38:5: nbqa black wrongpath/notebook.ipynb --nbqa-mutate
Oh no! 💥 💔 💥
1 file failed to reformat.
```

I would expect nbQA to warn about the missing file, but not fail because of it.

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #399

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
failed CI job is likely due to codecov being down :cry:

File: nbqa/output_parser.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    str
+        Stderr with references to temporary Python file's lines replaced with references
+        to notebook's cells and lines.
+    """
+
+    def substitution(match: Match[str]) -> str:
+        """Replace Python line with corresponding Jupyter notebook cell."""
+        return str(cell_mapping[int(match.group())])
+
+    pattern = _get_pattern(notebook, command)
```
Comment (girip11) [inline nbqa/output_parser.py:98]: How about something like returning a list of patterns for a command as well as its replacements too and applying all those patterns sequentially one after the other on out and err?

```Python
def _get_pattern(command: str):
      # here we will have a dict for each command
      # replacement could be a callable or a string(or whatever re.sub works with)
      black = [("pattern1", replacement1), ("pattern2", replacement2)]
      doctest = [("pattern1", replacement1), ("pattern2", replacement2)]

# In map_python_line_to_nb_lines
patterns = _get_patterns(command)
for pattern in patterns:
      out = re.sub(pat[0], pat[1], out)
```

Something one the above lines?

**Review by girip11 — CHANGES_REQUESTED** [review]

File: nbqa/output_parser.py (hunk 1 above)
Comment (MarcoGorelli) [reply nbqa/output_parser.py]: That would be cleaner, thanks! I done something along those lines

File: nbqa/output_parser.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        Code quality tool.
+    cell_mapping
+        Mapping from Python file lines to Jupyter notebook cells.
+
+    Returns
+    -------
+    List
+        Patterns and substitutions for reported output.
+    """
+    standard_substitution = partial(_line_to_cell, cell_mapping=cell_mapping)
```
Comment (girip11) [inline nbqa/output_parser.py:34]: This change looks much cleaner now.

**Review by girip11 — APPROVED** [review]

**Pull request merged**